clouddriver-kubernetes/src/integration/java/com/netflix/spinnaker/clouddriver/kubernetes/it/InfrastructureIT.java

```diff
@@ -829,4 +831,345 @@ public void shouldGetRawResources() throws InterruptedException, IOException {
+    // ------------------------- given --------------------------
+
+    String appName = "getmanifest";
+    System.out.println("> Using namespace " + ns);
+
+    KubeTestUtils.deployIfMissing(
+        baseUrl(), ACCOUNT1_NAME, ns, "networkPolicy", NETWORK_POLICY_1_NAME, appName, kubeCluster);
+    ;
+
+    KubeTestUtils.repeatUntilTrue(
+        () -> {
+          // ------------------------- when --------------------------
+          System.out.println("> Sending get securityGroups request");
+          Response resp = get(baseUrl() + "/securityGroups/" + ACCOUNT1_NAME + "/kubernetes");
+
+          // ------------------------- then --------------------------
+          System.out.println(resp.asString());
+          if (resp.statusCode() == 404) {
+            return false;
+          }
+          resp.then().statusCode(200);
+          List securityGroupList = resp.jsonPath().getList(ns);
+          return securityGroupList != null && securityGroupList.size() > 0;
+        },
+        CACHE_TIMEOUT_MIN,
+        TimeUnit.MINUTES,
+        "Waited "
+            + CACHE_TIMEOUT_MIN
+            + " minutes for GET /securityGroups/"
+            + ACCOUNT1_NAME
+            + "/kubernetes"
+            + " to return valid data");
+  }
+
+  @DisplayName(
+      ".\n===\n"
+          + "Given two kubernetes network policies for one account\n"
+          + "When sending get securityGroups/{account}/{cloudprovider}/{name}\n"
+          + "Then response should contain the security group specified in name\n===")
+  @Test
+  public void shouldListSecurityGroupsByAccountAndCloudProviderAndName()
+      throws InterruptedException, IOException {
+    // ------------------------- given --------------------------
+    String appName = "getmanifest";
+    System.out.println("> Using namespace " + ns);
+
+    KubeTestUtils.deployIfMissing(
+        baseUrl(), ACCOUNT1_NAME, ns, "networkPolicy", NETWORK_POLICY_1_NAME, appName, kubeCluster);
+    KubeTestUtils.deployIfMissing(
+        baseUrl(), ACCOUNT1_NAME, ns, "networkPolicy", NETWORK_POLICY_2_NAME, appName, kubeCluster);
+
+    KubeTestUtils.repeatUntilTrue(
+        () -> {
+          // ------------------------- when --------------------------
+          System.out.println("> Sending get securityGroups request");
+          Response resp =
+              get(
+                  baseUrl()
+                      + "/securityGroups/"
+                      + ACCOUNT1_NAME
+                      + "/kubernetes/networkpolicy "
+                      + NETWORK_POLICY_1_NAME);
+
+          // ------------------------- then --------------------------
+          System.out.println(resp.asString());
+          if (resp.statusCode() == 404) {
+            return false;
+          }
+          resp.then().statusCode(200);
+          List securityGroupList = resp.jsonPath().getList(ns);
+          return securityGroupList != null && securityGroupList.size() != 0;
```

Comment:
If you're requesting security groups by network policy name then the returned list should be exactly one item